### DISCUSSION TITLE
Correlate chat history with user email?

### DISCUSSION THREAD

**Discussion by tbendien** [open-webui/open-webui#1042] [Ideas]
@tjbck @justinh-rahb when exporting chat history, the email of the user is not included. 

How would I export chat history and correlate the user UUID with the user email please?

**Comment by justinh-rahb** [comment]
Would probably need to dump the entire database and then use some SQLite tools on it to do what I'm guessing you're doing (monitoring activity?), but I'm just guessing.

**Reply by tbendien** [reply]
Yep, I need to query the chat logs and correlate user ID's with chat history for sentiment analysis and Cyber audits. So need a way to export user history and put it into a DB like postgres.

**Reply by justinh-rahb** [reply]
Well, it's already natively a SQLite database, and there's a recently added button to download it in **Admin Settings**:
<img width="500" alt="Screenshot 2024-03-05 at 12 04 45 AM" src="https://github.com/open-webui/open-webui/assets/52832301/bce97470-d6f5-421d-a97f-35adb9ecb79e">

That's not even the way I'd recommend though, because you could also bind-mount the data volume to the host, and use a SQLite client to _read_ it directly and in real-time and skip the downloading.

**Reply by tbendien** [reply]
Hi @justinh-rahb  how would I connect to the SQL Lite DB so I can run realtime queries against the DB? I assume its something like <hostip:port> with DBEaver or some such tool? I was able to download the DB and figured out how to use DB BRowser for SQLite to browse the data, but now want to query the DB real time without downloading it.. Thanks in advance for your help!
